cloudwatch_monitoring/lambdas.py

```diff
@@ -234,38 +238,59 @@ def create_lambda_widgets(region, deploy_stage):
                 }
             }
 
+            memory_usage_widget = {
+                "type": "log",
+                'height': positioning['height'],
+                'width': positioning['width'],
+                "properties": {
+                    "query": f"SOURCE '/aws/lambda/{function_name}' | filter @type=\"REPORT\" | avg(@maxMemoryUsed) as mean_MemoryUsed, max(@maxMemoryUsed) as max_MemoryUsed by bin(5min)",
+                    "region": region,
+                    "title": f"{widget_title} Memory Usage",
+                    "view": "timeSeries",
+                    "stacked": False
+                }
+            }
+
             if 'dv' == widget_etl_branch:
                 dv_widgets.append(widget)
                 dv_widgets.append(concurrent_executions_widget)
                 dv_widgets.append(duration_widget)
+                dv_widgets.append(memory_usage_widget)
             elif 'sv' == widget_etl_branch:
                 sv_widgets.append(widget)
                 sv_widgets.append(concurrent_executions_widget)
                 sv_widgets.append(duration_widget)
+                sv_widgets.append(memory_usage_widget)
             elif 'environment_management' == widget_etl_branch:
                 environment_management_widgets.append(widget)
                 environment_management_widgets.append(concurrent_executions_widget)
                 environment_management_widgets.append(duration_widget)
+                environment_management_widgets.append(memory_usage_widget)
             elif 'error_handling' == widget_etl_branch:
                 error_widgets.append(widget)
                 error_widgets.append(concurrent_executions_widget)
                 error_widgets.append(duration_widget)
+                error_widgets.append(memory_usage_widget)
             elif 'data_ingest' == widget_etl_branch:
                 data_in_widgets.append(widget)
                 data_in_widgets.append(concurrent_executions_widget)
                 data_in_widgets.append(duration_widget)
+                data_in_widgets.append(memory_usage_widget)
             elif 'data_purging' == widget_etl_branch:
                 data_purge_widgets.append(widget)
                 data_purge_widgets.append(concurrent_executions_widget)
                 data_purge_widgets.append(duration_widget)
+                data_purge_widgets.append(memory_usage_widget)
             elif 'nwis_web' == widget_etl_branch:
                 nwis_web_widgets.append(widget)
                 nwis_web_widgets.append(concurrent_executions_widget)
                 nwis_web_widgets.append(duration_widget)
+                nwis_web_widgets.append(memory_usage_widget)
             else:
                 misc_widgets.append(widget)
                 misc_widgets.append(concurrent_executions_widget)
                 misc_widgets.append(duration_widget)
+                misc_widgets.append(memory_usage_widget)
```

Comment:
The list of appends has been increasing. I'm wondering if it's time to consider an "internal function" that does this heavy-lifting.

Reply:
Agreed, on it.

Reply:
Actually, let me address this in a separate merge.

Reply:
@ayan-usgs this commit addresses this issue: https://github.com/usgs/aqts-capture-cloudwatch-dashboard/pull/17/commits/bf0dbb7c9d86c3067a3e11221d29fc5bef07d26e#diff-87638053009b1a01dd8e1083c1e380d324408d09057d8ad300e151f8a2b6f0ffR257